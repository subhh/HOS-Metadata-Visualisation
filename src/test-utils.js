import React, { useReducer } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { StateContext, DispatchContext, reducer, initialState } from './state'
import theme from './theme'

const AllTheProviders = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <ThemeProvider theme={theme}>
      <StateContext.Provider value={state}>
        <DispatchContext.Provider value={dispatch}>
          {children}
        </DispatchContext.Provider>
      </StateContext.Provider>
    </ThemeProvider>
  )
}

const WithRouter = ({ children }) => (
  <MemoryRouter>
    <AllTheProviders>{children}</AllTheProviders>
  </MemoryRouter>
)

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderWithRouter = (ui, options) =>
  render(ui, {
    wrapper: WithRouter,
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const mockAsync = data => {
  jest.spyOn(global, 'fetch').mockImplementation(url => {
    return Promise.resolve({
      json: () => Promise.resolve(data),
    })
  })
}

export const restoreAsyncMock = () => {
  global.fetch.mockRestore()
}
