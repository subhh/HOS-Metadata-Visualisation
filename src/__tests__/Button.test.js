import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from 'test-utils'
import Button from 'components/ButtonLink'

it('<Button /> should match snapshot', () => {
  const { container } = render(
    <MemoryRouter>
      <Button to="somewhere">Button</Button>
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})
