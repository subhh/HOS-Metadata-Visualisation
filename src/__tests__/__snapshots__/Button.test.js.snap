// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <a
    class="sc-htoDjs bZzhDu"
    href="/somewhere"
  >
    <button
      class="sc-bdVaJa sc-EHOje sc-dnqmqq cRFAdu"
      color="blueMediumDark"
      font-size="inherit"
      font-weight="bold"
    >
      Button
    </button>
  </a>
</div>
`;
